### ISSUE TITLE
How can/should I license these word lists?

### ISSUE THREAD

**Issue by sts10** [sts10/orchard-street-wordlists#1]
It's my understanding that all of the text that makes up Wikipedia is [dual-licensed as both Creative Commons Attribution-ShareAlike 3.0 Unported License (“CC BY-SA”), and GNU Free Documentation License (“GFDL”)](https://foundation.wikimedia.org/wiki/Policy:Terms_of_Use#7._Licensing_of_Content). Note that technically this project uses data from dumps.wikimedia.org, which seems to have [the same licensing terms](https://dumps.wikimedia.org/legal.html).

Since all of the Orchard Street Wordlists use Wikipedia text data (they are partially comprised of frequently used words from Wikipedia), and since [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/) since it is a "ShareALike" license, I have licensed all of the Orchard Street Wordlists under CC BY-SA 3.0. (I've heard mixed things about GFDL...)

**However**, I'd _like_ to license the lists under the newer [CC BY-SA version 4.0 license](http://creativecommons.org/licenses/by/4.0/), to take advantage of the improvements in the 4.0 suite of licenses, as Creative Commons recommends. 

My question is: Would licensing these word lists under CC BY-SA 4.0 violate Wikipedia's CC BY-SA 3.0 licensing? Creative Commons has [a guide to upgrading to 4.0](https://wiki.creativecommons.org/wiki/4.0_upgrade_guidelines), which seems to imply that I can't just bump from 3.0 to 4.0, but I'm not sure. I may be able to dual-license under both 3.0 and 4.0? I welcome input here!

**Comment by sts10** [comment]
Interestingly, Wikimedia/Wikipedia itself seems to be [proposing an upgrade from CC BY-SA 3.0 to CC BY-SA 4.0](https://meta.wikimedia.org/wiki/Wikimedia_Foundation_Legal_department/2023_ToU_updates/Proposed_update#7._Licensing_of_Content). 

Perhaps tellingly of the stickiness of the situation, there is [some discussion about whether this will work legally](https://meta.wikimedia.org/wiki/Talk:Terms_of_use#Change_to_CC_BY-SA_4.0).

**Comment by sts10** [comment]
Maybe section 4 b of [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/legalcode) saves us?!

> You may Distribute or Publicly Perform an Adaptation only under the terms of: (i) this License; (ii) a later version of this License with the same License Elements as this License; (iii) a Creative Commons jurisdiction license (either this or a later license version) that contains the same License Elements as this License (e.g., Attribution-ShareAlike 3.0 US)); (iv) a Creative Commons Compatible License.

Obviously CC BY-SA 4.0 is a later version of 3.0. But can we safely assume that 4.0 has "the same License Elements as [3.0]"?

**Comment by sts10** [comment]
I was about to be ready to conclude I have to stick to 3.0, but then I found two interesting links:

Citing Section 4(b) in the [CC BY-SA 3.0 version of the license](https://creativecommons.org/licenses/by-sa/3.0/legalcode) (as I do above), [Wikimedia explains that](https://meta.wikimedia.org/wiki/Terms_of_use/Creative_Commons_4.0/Legal_note#Mechanism_for_the_upgrade):

> This allows people to submit contributions under the terms of a later version of a Creative Commons Attribution-ShareAlike license. In other words, [CC BY-SA 3.0 is compatible](https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses/) with CC BY-SA 4.0 and later versions of the same license. After the Terms of Use are modified to use version 4.0, an article may be used under the 4.0 version of the license according to the [attribution requirements in the Wikimedia Terms of Use](https://foundation.wikimedia.org/wiki/Terms_of_use#7._Licensing_of_Content).

(The page linked in the paragraph above has [Creative Commons itself explicitly says that CC BY-SA 3.0 is compatible with CC BY-SA 4.0](https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses/), which is good to know for sure!)

I'll note that I've left [a comment on the discussion of Wikipedia's update from 3.0 to 4.0](https://meta.wikimedia.org/wiki/Talk:Terms_of_use#c-Survive_morse_mocker-20230414220800-Pajz-20230316225600), seeking both general clarification and advice on my licensing choices of this project. 

I'm (still) not sure if I can just slap a CC 4.0 on this project, but I'm gathering more evidence!

**Comment by sts10** [comment]
Another piece of evidence in favor of "Yes, we're legally able to switch this project to CC 4.0 BY-SA if we so please" from [section 7h of the (current) Wikimedia Terms of use](https://meta.wikimedia.org/wiki/Terms_of_use#7._Licensing_of_Content): 

> Modifications or additions to material that you re-use: When modifying or making additions to text that you have obtained from a Project website, you agree to license the modified or added content under CC BY-SA 3.0 or later (or, as explained above, another license when exceptionally required by the specific Project edition or feature).

I'd guess that creating a word list with Wikipedia text data qualifies as "modifications" here?

**Comment by sts10** [comment]
We should also ask if an alphabetized word list, such as [the Orchard Street Long List](https://github.com/sts10/orchard-street-wordlists/blob/main/lists/orchard-street-long.txt), is original enough to even be copyrighted in the first place. That all of my lists are in alphabetical order likely helps the argument that they are not original enough to copyright.

Clearly everything is not copyrightable, even under Creative Commons. I can’t write a very simple programming function that, say, lower-cases any string given to it and then license it under GPL.

Interestingly, the [EFF fandom word lists](https://www.eff.org/deeplinks/2018/08/dragon-con-diceware) are licensed under Creative Commons Attribution 3.0 United States ([proof](https://www.eff.org/files/2018/08/29/gameofthrones_8k-2018.txt)). 

Ironically, I wouldn’t be upset if I could not copyright my word lists. While my pride might be a little hurt, it would make sharing the lists easier.

## Next steps?

In practical terms, I think replacing the current CC BY-SA 3.0 Unported license with the [CC0 license](https://creativecommons.org/share-your-work/public-domain/cc0/) would be a proper way to show/assert/embrace this lack of originality and thus lack of copyright.

Unfortunately I just don’t have enough information to know exactly what to do. Leaving the CC BY-SA 3.0 license in place feels safe but at the same time part of me wants to slap the CC0 on there and not think about this anymore!

**Comment by sts10** [comment]
Sometime earlier this month, [Wikipedia switched the license for its text](https://foundation.wikimedia.org/w/index.php?title=Policy:Terms_of_Use&oldid=256875#7._Licensing_of_Content) from CC BY-SA 3.0 to CC BY-SA 4.0. 

I ran a fresh scrape of Wikipedia word frequency data, then re-created all of the Orchard Street Wordlists using that word data (still combined with Google Ngram data). Thus, I felt cleared to re-license these new lists under CC BY-SA 4.0. I look for to enjoying the advantages of a CC 4.0 license for these lists. 

Closing this issue for now.
